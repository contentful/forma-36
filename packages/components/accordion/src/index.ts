export { Accordion } from './Accordion';
export type { AccordionProps } from './Accordion';
export { AccordionHeader } from './AccordionHeader/AccordionHeader';
export type { AccordionHeaderProps } from './AccordionHeader/AccordionHeader';
export { AccordionItem } from './AccordionItem/AccordionItem';
export type { AccordionItemProps } from './AccordionItem/AccordionItem';
export { AccordionPanel } from './AccordionPanel/AccordionPanel';
export type { AccordionPanelProps } from './AccordionPanel/AccordionPanel';
