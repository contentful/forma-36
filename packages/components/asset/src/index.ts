export { Asset } from './Asset';
export type { AssetProps } from './Asset';
