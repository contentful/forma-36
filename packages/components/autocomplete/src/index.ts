export { Autocomplete } from './Autocomplete';
export type { AutocompleteProps } from './Autocomplete';
