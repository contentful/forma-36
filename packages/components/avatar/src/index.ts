export { Avatar } from './Avatar';
export type { AvatarProps } from './Avatar';
export { AvatarGroup } from './AvatarGroup';
export type { AvatarGroupProps } from './AvatarGroup';
