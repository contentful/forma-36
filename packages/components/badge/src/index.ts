export { Badge } from './Badge';
export type { BadgeProps } from './Badge';
export { EntityStatusBadge } from './EntityStatusBadge';
export type { EntityStatusBadgeProps } from './EntityStatusBadge';
export type { BadgeVariant } from './types';
