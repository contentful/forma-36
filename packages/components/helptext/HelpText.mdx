---
title: 'HelpText'
type: 'component'
status: 'stable'
slug: /components/help-text/
github: 'https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components/src/components/HelpText'
storybook: 'https://f36-storybook.contentful.com/?path=/story/components-helptext--default'
typescript: ./src/HelpText.tsx
---

`HelpText` is a styled copy block with guidance, placed in the context of form components.

## Table of contents

- [How to use HelpText](#how-to-use-HelpText)
- [Code example](#code-example)
- [Content recommendations](#content-recommendations)
- [Props](#props)

## How to use HelpText

- Only use this component in the context of a form, and as corresponding guidance to an input field

## Code example

```jsx
<React.Fragment>
  <FormLabel htmlFor="name">Full name</FormLabel>
  <TextInput
    name="name"
    value="Claus Mitchell"
    className="f36-margin-bottom--xs"
  />
  <HelpText>Enter your first and last name</HelpText>
</React.Fragment>
```

## Content recommendations

- Use direct, succinct copy that helps the user to successfully complete the form
- Do not preface the instructions with introductory text, such as "please"
- To add additional context, link to documentation
- Do not use punctuation
- Use sentence style caps (in which only the first word of a sentence or term is capitalized)

## Props

import { Props } from '@contentful/f36-docs-utils';

<Props of="HelpText" />
