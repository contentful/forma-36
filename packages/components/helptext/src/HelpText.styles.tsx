import { css } from 'emotion';
import tokens from '@contentful/f36-tokens';

export const getStyles = () => ({
  root: css({
    marginBottom: 0,
    marginTop: 0,
    fontFamily: tokens.fontStackPrimary,
    color: tokens.gray500,
    fontSize: tokens.fontSizeM,
    lineHeight: tokens.lineHeightDefault,
    textRendering: 'optimizeLegibility',
  }),
});
