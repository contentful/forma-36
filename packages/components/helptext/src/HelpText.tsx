import React from 'react';
import { CommonProps, PropsWithHTMLElement } from '@contentful/f36-core';
import { Text } from '@contentful/f36-typography';

export type HelpTextInternalProps = CommonProps & {
  children: React.ReactNode;
};

export type HelpTextProps = PropsWithHTMLElement<HelpTextInternalProps, 'p'>;

/**
 * `HelpText` is a styled copy block with guidance, placed in the context of form components.
 */

export const HelpText = React.forwardRef<HTMLParagraphElement, HelpTextProps>(
  ({ testId = 'cf-ui-help-text', ...otherProps }, ref) => {
    return (
      <Text
        as="p"
        fontColor="gray500"
        fontSize="fontSizeM"
        lineHeight="lineHeightDefault"
        testId={testId}
        {...otherProps}
        ref={ref}
      />
    );
  },
);

HelpText.displayName = 'HelpText';
