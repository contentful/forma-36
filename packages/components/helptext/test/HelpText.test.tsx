import React from 'react';
import { render } from '@testing-library/react';
import { axe } from '@/scripts/test/axeHelper';
import { HelpText } from '../src/HelpText';

describe('HelpText', function () {
  it('renders the component', () => {
    const { container } = render(
      <HelpText>Lorem Ipsum dolor sit amet</HelpText>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the component with an additional class name', () => {
    const { container } = render(
      <HelpText className="my-extra-class">
        Lorem Ipsum dolor sit amet
      </HelpText>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('has no a11y issues', async () => {
    const { container } = render(
      <HelpText>Lopem Ipsum dolor sit amet</HelpText>,
    );
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
