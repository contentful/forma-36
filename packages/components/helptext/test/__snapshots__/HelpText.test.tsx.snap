// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpText renders the component 1`] = `
.emotion-0 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  color: #67728A;
  font-size: 0.875rem;
  line-height: 1.5;
  text-rendering: optimizeLegibility;
}

<p
  class="emotion-0"
  data-test-id="cf-ui-help-text"
>
  Lorem Ipsum dolor sit amet
</p>
`;

exports[`HelpText renders the component with an additional class name 1`] = `
.emotion-0 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  color: #67728A;
  font-size: 0.875rem;
  line-height: 1.5;
  text-rendering: optimizeLegibility;
}

<p
  class="my-extra-class emotion-0"
  data-test-id="cf-ui-help-text"
>
  Lorem Ipsum dolor sit amet
</p>
`;
